src/core/compact_object.cc

```diff
@@ -811,6 +810,13 @@ void CompactObj::SetJsonSize(int64_t size) {
   }
 }
 
+void CompactObj::SetStreamSize(int64_t size) {
+  if (size < 0) {
+    DCHECK(static_cast<int64_t>(u_.r_obj.Size()) >= size);
```

Comment:
Can you add a comment explaining this? I don't understand, and a negative size isn't intuitive to me

Reply:
Yes! It's similar to Json, maybe we deallocate some part of the stream (let's say because you deleted a consumer), so the net will be negative.

Reply:
Now I realize that this function accepts a **delta** and not a real size. I'd strongly recommend renaming it to reflect that. Definitely the argument (call it `delta`), but also the name (maybe `AddStreamSize()`?)

Reply:
Yes or `IncrementStreamSize` although `Increment` means "increasing" and it;s not very accurate :smile:

Reply:
Indeed increment, to me, sounds like it's ever increasing. You could `Add()` negative numbers though